### PR TITLE
Add support for the --ignore-plugins flag

### LINKED ISSUE #77: Add --ignore-plugins flag to abell serve
**Is your feature request related to a problem? Please describe.**

Plugins can have additional sources (CHANGELOGS, READMES) which can slow down the development process if you don't want to test them on every run.

**Describe the solution you'd like**

Calling `abell serve --ignore-plugins` should ignore the plugin execution.

**Approach**

1. We will have to register this flag in [bin/abell.js](https://github.com/abelljs/abell/blob/main/bin/abell.js#L22)
2. Add flag value `true` or `false` to `programInfo` in [src/commands/serve.js](https://github.com/abelljs/abell/blob/main/src/commands/serve.js#L245)
3. On true, ignore calling `executeAfterBuildPlugins` and `executeBeforeBuildPlugins` functions in the serve.js
4. ignore calling `executeBeforeHTMLWritePlugins` from [src/utils/generate-site.js](https://github.com/abelljs/abell/blob/312dd6a057ce0d39142b803ad220798706137e9e/src/utils/generate-site.js#L154)

This issue can be a good start for beginners so if you're experienced and comfortable with GitHub, Please look for other issues and leave this issue for beginners to that they can get started :D

### PR DESCRIPTION
**Pull request by abhijit-hota**
## What type of PR is this?

- [x] Feature
- [ ] Bug Fix
- [ ] Refactor
- [ ] Documentation Update

## Description
Implemented the `--ignore-plugins` flag for `abell serve`
Resolves #77 

Hope this is good enough. Passed all the tests and smoke tested it on my own. 
## Added unit tests?

- [x] No, I want help in writing tests.
- [ ] No, I want someone else to write tests.
- [ ] No, The addition does not need tests.
- [ ] Yes

## [optional] Random GIF to +1 someone's day

Here is pigeon band in hoodies by one of my favorite artists: [Keke](https://k-eke.tumblr.com/)

### CODE REVIEW

**Comment by abhijit-hota** [comment]
Sorry for the typo in the commit message ( ~`--ignore-plugin`~`--ignore-plugins`).

Also can I do something about the one failing test?

File: package.json [hunk 1]
```diff
@@ -65,4 +65,4 @@
       "pre-commit": "npm run prettier && eslint . --fix && npm run eslint"
     }
   }
-}
\ No newline at end of file
+}
```
Comment (abhijit-hota) [inline package.json]: I don't what made this to change. Also, the version was bumped to `0.6.4` in `package-lock.json`.

File: src/utils/generate-site.js [hunk 2]
```diff
@@ -150,8 +150,10 @@ async function createHTMLFile(templateObj, programInfo, options) {
     });
   }
 
-  // Execute beforeHTMLWrite plugins
-  htmlOut = await executeBeforeHTMLWritePlugins(htmlOut, programInfo);
+  // Execute beforeHTMLWrite plugins if the --ignore-plugins flag is not passed
+  if (!programInfo.ignorePlugins) {
```
Comment (abhijit-hota) [inline src/utils/generate-site.js:154]: Seems like an end of line issue with my file?
I'll be glad to help after someone reviews it.

**Comment by abhijit-hota** [comment]
So the linting was failing for me locally too. I added the below config to the `.eslintrc.js` file and it passed.
As I suspected, it was an endOfLine issue (I'm on a Windows machine).
```js
...
    'prettier/prettier': [
      'error',
      {
        endOfLine: 'auto'
      }
    ],
...
```
I can commit the changes if you want to. 
I have run `npm run prettier` and `npm run eslint`.

File: package.json (hunk 1 above)
Comment (saurabhdaware) [reply package.json]: yeah that's normal. Its fine

File: src/utils/generate-site.js (hunk 2 above)
Comment (saurabhdaware) [reply src/utils/generate-site.js:154]: I'll just check locally and let you know

Comment (saurabhdaware) [reply src/utils/generate-site.js]: I pushed a change. It should fix it. In future if you come across an eslint issue, you can fix it with `npx eslint . --fix`.

**Review by saurabhdaware — APPROVED** [review]
Thank you so much @abhijit-hota! I tested it locally and it works great! Before we test this we will also have to write tests for dev-server so I'll do it later. 

Thank you so much! Releasing this in v0.6.5

**Comment by abhijit-hota** [comment]
> Thank you so much @abhijit-hota! I tested it locally and it works great! Before we test this we will also have to write tests for dev-server so I'll do it later.
> 
> Thank you so much! Releasing this in v0.6.5

Glad! 😄

**Pull request merged**